code/modules/mob/living/carbon/carbon.dm

```diff
@@ -616,4 +611,289 @@ var/const/GALOSHES_DONT_HELP = 4
 /mob/living/carbon/fully_replace_character_name(oldname,newname)
 	..()
 	if(dna)
-		dna.real_name = real_name
\ No newline at end of file
+		dna.real_name = real_name
+
+/mob/living/carbon/update_sight()
+	if(!client)
+		return
+	if(stat == DEAD)
+		sight = (SEE_TURFS|SEE_MOBS|SEE_OBJS)
+		see_in_dark = 8
+		see_invisible = SEE_INVISIBLE_OBSERVER
+		return
+
+	see_invisible = initial(see_invisible)
+	see_in_dark = initial(see_in_dark)
+	sight = initial(sight)
+
+	if(client.eye != src)
+		var/atom/A = client.eye
+		if(A.update_remote_sight(src)) //returns 1 if we override all other sight updates.
+			return
+
+	for(var/obj/item/organ/internal/cyberimp/eyes/E in internal_organs)
+		sight |= E.sight_flags
+		if(E.dark_view)
+			see_in_dark = max(see_in_dark,E.dark_view)
+		if(E.see_invisible)
+			see_invisible = min(see_invisible, E.see_invisible)
+
+	if(see_override)
+		see_invisible = see_override
+
+
+//to recalculate the mob's total tint from tinted equipment it's wearing.
+/mob/living/carbon/proc/update_tinttotal()
+	if(!tinted_weldhelh)
+		return
+	tinttotal = 0
+	if(istype(head, /obj/item/clothing/head))
+		var/obj/item/clothing/head/HT = head
+		tinttotal += HT.tint
+	if(wear_mask)
+		tinttotal += wear_mask.tint
+	update_vision_overlays()

```

Comment:
This fixes monkeys not getting the same vision impairment as humans when wearing welding helmet.